src/main/java/com/example/onlineservicesemulator/Controller.java

```diff
@@ -0,0 +1,59 @@
+package com.example.onlineservicesemulator;
+
+import com.example.onlineservicesemulator.classes.JSONReader;
+import javafx.event.EventHandler;
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.ListView;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.layout.VBox;
+
+import java.net.URL;
+import java.util.*;
+
+public class Controller implements Initializable {
+
+    @FXML
+    public ListView<String> filesList;
+    @FXML
+    private VBox checkboxList;
+    ArrayList<String> servicesNames;
+    Map<String, List<String>> uploadedFiles;
+
+    @Override
+    public void initialize(URL url, ResourceBundle resourceBundle) {
+        servicesNames = JSONReader.getServices();
+        setFilesMap();
+        setCheckboxList();
```

Comment:
create new method setCheckboxListeners()
and move the logic from setCheckboxList to setCheckboxListeners for creating the listeners